### ISSUE TITLE
When using Lit Element, CloudFlare function cannot be published because HTMLElement is not defined

### ISSUE THREAD

**Issue by dengel29** [withastro/adapters#141]
### Astro Info

```block
Astro                    v4.1.3
Node                     v19.9.0
System                   macOS (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/lit
```

### Describe the Bug

I am trying to deploy a project using Lit on Cloudflare Pages using the @astrojs/cloudflare adapter. Locally, everything is working fine, but when pushing to Cloudflare pages using the Cloudflare adapter, I get the following error message after a successful build:

`Error: Failed to publish your Function. Got error: Uncaught ReferenceError: HTMLElement is not defined
  at functionsWorker-0.056383667617412536.js:1471:3`
  
  I get several messages further up the stacktrace that give me a warning:
  
  `Ambiguous external namespace resolution: "node_modules/.pnpm/lit@3.1.1/node_modules/lit/index.js" re-exports "html" from one of the external modules "lit-element/lit-element.js" and "lit-html/is-server.js", guessing "lit-element/lit-element.js".****`
  
  This makes me worried that the DOM-shim provided by the @lit-labs/ssr is not being included in the bundle. Is it possible to fix this ambiguous namespace, or a direct way to load the DOM shim so HTMLElement is recognized in the Cloudflare Function?
  
 the `astro.config.mjs`:
 ```mjs
import { defineConfig } from 'astro/config';
import lit from "@astrojs/lit";
import cloudflare from "@astrojs/cloudflare";

// https://astro.build/config
// @see https://developers.cloudflare.com/pages/framework-guides/deploy-an-astro-site/#modes
export default defineConfig({
  integrations: [lit()],
  output: "server", // both outputs give same error
  adapter: cloudflare({mode: 'directory'}) // both modes give same error
});
 ```
 
 the `package.json`:
 ```json
 {
  "main": "index.js",
  "scripts": {
    "dev": "astro dev",
    "start": "astro dev",
    "build": "astro build",
    "preview": "astro preview"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "@astrojs/cloudflare": "^8.1.0",
    "@astrojs/lit": "^4.0.1",
    "@astrojs/mdx": "^2.0.4",
    "@lit-labs/ssr": "^3.2.1",
    "@lit-labs/ssr-dom-shim": "^1.1.2",
    "@webcomponents/template-shadowroot": "^0.2.1",
    "astro": "^4.1.3",
    "lit": "^3.1.1",
    "open-props": "^1.6.17",
    "@cloudflare/workers-types": "^4.20240117.0",
  }
}
```

I found these related issues + PRs: 
* https://github.com/withastro/adapters/issues/42 which was closed in favor of this less stale issue https://github.com/withastro/adapters/issues/56 
* https://github.com/withastro/astro/pull/6915 
 
 
 Here is the more complete stacktrace from the Cloudflare console:
 
 <pre class="c_c c_bw c_oi c_jk c_aci c_acj c_dc c_zc c_acw c_acx c_pl c_fa c_fb c_pf c_by c_acy">
10:40:10.637 | Ambiguous external namespace resolution: "node_modules/.pnpm/lit@3.1.1/node_modules/lit/index.js" re-exports "css" from one of the external modules "lit-element/lit-element.js" and "lit-html/is-server.js", guessing "lit-element/lit-element.js".
-- | --
10:40:10.638 | Ambiguous external namespace resolution: "node_modules/.pnpm/lit@3.1.1/node_modules/lit/index.js" re-exports "html" from one of the external modules "lit-element/lit-element.js" and "lit-html/is-server.js", guessing "lit-element/lit-element.js".
10:40:10.639 | Ambiguous external namespace resolution: "node_modules/.pnpm/lit@3.1.1/node_modules/lit/index.js" re-exports "LitElement" from one of the external modules "lit-element/lit-element.js" and "lit-html/is-server.js", guessing "lit-element/lit-element.js".
10:40:10.886 | "css", "html" and "LitElement" are imported from external module "lit-html/is-server.js" but never used in "node_modules/.pnpm/lit@3.1.1/node_modules/lit/index.js".
10:40:10.950 | 02:40:10 [build] ✓ Completed in 1.05s.
10:40:10.950 |  
10:40:10.951 | building client (vite)
10:40:10.960 | 02:40:10 [vite] transforming...
10:40:11.063 | 02:40:11 [vite] ✓ 33 modules transformed.
10:40:11.082 | 02:40:11 [vite] rendering chunks...
10:40:11.089 | 02:40:11 [vite] computing gzip size...
10:40:11.092 | 02:40:11 [vite] dist/_astro/directive.xgBC_cM0.js                          0.42 kB │ gzip: 0.31 kB
10:40:11.093 | 02:40:11 [vite] dist/_astro/client.KliVgTTX.js                             0.62 kB │ gzip: 0.39 kB
10:40:11.093 | 02:40:11 [vite] dist/_astro/HeaderNav.0BeSTiyq.js                          2.13 kB │ gzip: 0.97 kB
10:40:11.093 | 02:40:11 [vite] dist/_astro/XCheckbox.y_7YmnQg.js                          2.89 kB │ gzip: 1.38 kB
10:40:11.093 | 02:40:11 [vite] dist/_astro/astro_scripts/before-hydration.js.cW-K1qaT.js  5.64 kB │ gzip: 2.36 kB
10:40:11.093 | 02:40:11 [vite] dist/_astro/lit-html.t5Xi0LsG.js                           7.17 kB │ gzip: 3.16 kB
10:40:11.093 | 02:40:11 [vite] dist/_astro/custom-element.TrIwOAjq.js                     8.19 kB │ gzip: 3.02 kB
10:40:11.094 | 02:40:11 [vite] ✓ built in 138ms
10:40:11.142 | 02:40:11
10:40:11.142 | finalizing server assets
10:40:11.142 |  
10:40:11.143 | 02:40:11 [build] Rearranging server assets...
10:40:11.228 | 02:40:11 [build] Server built in 1.43s
10:40:11.228 | 02:40:11 [build] Complete!
10:40:11.271 | Finished
10:40:11.272 | Found Functions directory at /functions. Uploading.
10:40:12.554 | ✨ Compiled Worker successfully
10:40:12.584 | Found _routes.json in output directory. Uploading.
10:40:12.596 | Validating asset output directory
10:40:13.602 | Deploying your site to Cloudflare's global network...
10:40:18.373 | Uploading... (8/8)
10:40:18.374 | ✨ Success! Uploaded 0 files (8 already uploaded) (0.67 sec)
10:40:18.374 |  
10:40:18.868 | ✨ Upload complete!
10:40:21.562 | Success: Assets published!
10:40:22.626 | Error: Failed to publish your Function. Got error: Uncaught ReferenceError: HTMLElement is not defined   at functionsWorker-0.056383667617412536.js:1471:3   at functionsWorker-0.056383667617412536.js:7:38   at functionsWorker-0.056383667617412536.js:1968:3   at functionsWorker-0.056383667617412536.js:7:38   at functionsWorker-0.056383667617412536.js:7338:3   at functionsWorker-0.056383667617412536.js:7:38   at functionsWorker-0.056383667617412536.js:7428:3   at functionsWorker-0.056383667617412536.js:7:38   at functionsWorker-0.056383667617412536.js:7435:3   at functionsWorker-0.056383667617412536.js:7:38</pre>

### What's the expected result?

I expected the Function to be able to be published after being built successfully. Everything local is working fine.

### Link to Minimal Reproducible Example

https://github.com/dengel29/dngl

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by alexanderniebuhr** [comment]
Thanks for the report. I'm not 100% sure but I do agree with @matthewp in the original PR. 
If this is something package specific, we will not add it, but we will allow users to customize some options soon.

**Comment by alexanderniebuhr** [comment]
You might also want to check out https://github.com/lit/lit/issues/3907 and https://github.com/lit/lit/issues/1983

**Comment by alexanderniebuhr** [comment]
This will be fixed by #56
